docs/book/src/component-config-tutorial/testdata/project/Makefile

```diff
@@ -11,12 +11,18 @@ else
 GOBIN=$(shell go env GOBIN)
 endif
 
+# GOFLAGS defines additional golang build flags like -tags and -ldflags
+GOFLAGS ?=
```

Comment:
Would we have any reason to use go flags by default? 
If so, what are the reasons? What are the flags?

Reply:
One of the most common use cases is with versioning e.g. `--ldflags="-X 'github.com/mycorp/mycorp-operator/version.Version=$(VERSION)'"`. There are other projects that pass along `-mod=vendor` or some other go compilation flags or passing tags during build to enable/disable feature builds. With the version example, maybe it also makes sense to scaffold it by default, but it might not.

Reply:
That is fine. 
but why we should incorporate those in the default scaffold?
will those be required for any use case?